main.go

```diff
@@ -19,105 +17,122 @@ type FileInfo map[string]time.Time
 var watchedFiles = FileInfo{}
 
 func main() {
-	defaultPath, _ := os.Getwd()
+	defaultPath, err := os.Getwd()
+	if err != nil {
+		log.Fatal("Getwd: ", err)
+	}
+
 	rootPath := flag.String("path", defaultPath, "the root path to be watched")
-	interval := flag.Int("interval", 5, "the interval (in seconds) for scanning files")
+	interval := flag.Duration("interval", 5*time.Second, "the interval (in seconds) for scanning files")
 	flag.Parse()
 
-	watchedFiles = walk(rootPath)
-	scheduleScanAt(rootPath, interval)
-}
+	if *interval < 0 {
+		log.Fatal("negative interval:", *interval)
+	}
 
-func scheduleScanAt(rootPath *string, interval *int) {
-	runTicker := time.NewTicker(
-		time.Duration(time.Duration(*interval) * time.Second))
-	for {
-		select {
-		case <-runTicker.C:
-			scan(rootPath)
-		}
+	if err := os.Chdir(*rootPath); err != nil {
+		log.Fatal("Chdir:", err)
+	}
+
+	watchedFiles = walk(rootPath)
+	for range time.NewTicker(*interval).C {
+		scan(rootPath)
 	}
 }
 
 func scan(rootPath *string) {
+	changedDirs := make(map[string]bool, 0)
```

Comment:
What do you think of changing `changed` to `modified`, so it stays more consistent with the `mod` of Go's file API?

```suggestion
	modifiedDirs := make(map[string]bool, 0)
```